app/src/main/java/net/gsantner/markor/frontend/textview/HighlightingEditor.java

```diff
@@ -116,24 +119,33 @@ private void updateDynamicHighlighting() {
         }
     }
 
+    private synchronized void postUpdateDynamicHighlighting() {
+        if (!_hlPostQueued) {
+            _hlPostQueued = true;
+            post(() -> {
+                updateDynamicHighlighting();
+                _hlPostQueued = false;
+            });
+        }
+    }
```

Comment:
Using the post prevents the selection from being hidden when the keyboard is shown